app/Http/Controllers/Front/ClientController.php

```diff
@@ -454,29 +470,25 @@ public function getPaymentByOrderIdClient($orderid, $userid)
     {
         try {
             $order = $this->order->where('id', $orderid)->where('client', $userid)->first();
-            $relation = $order->invoiceRelation()->pluck('invoice_id')->toArray();
-            if (count($relation) > 0) {
+            $relation = $order->invoiceRelation()->value('invoice_id');
+            if ($relation) {
                 $invoices = $relation;
             } else {
-                $invoices = $order->invoice()->pluck('id')->toArray();
+                $invoices = $order->invoice()->value('id');
             }
-            $payments = $this->payment->whereIn('invoice_id', $invoices)
-                    ->select('id', 'invoice_id', 'user_id', 'payment_method', 'payment_status', 'created_at', 'amount');
-            //dd(\Input::all());
-            return \DataTables::of($payments->get())
-                            ->addColumn('number', function ($model) {
-                                return $model->invoice()->first()->number;
+            $payments = Payment::leftJoin('invoices','payments.invoice_id','=', 'invoices.id')
```

Comment:
are you sure you want to use `leftjoin` here? products with no invoices too will be coming here

Reply:
We want to show payments for no invoices also in case if someone pays in advance before his invoice is generated